src/raftstore/store/worker/apply.rs

```diff
@@ -0,0 +1,1243 @@
+               index);
+
+        let cb = match cmd_cb {
+            None => return exec_result,
+            Some(cb) => cb,
+        };
+
+        // TODO: if we have exec_result, maybe we should return this callback too. Outer
+        // store will call it after handing exec result.
+        // Bind uuid here.
+        cmd_resp::bind_uuid(&mut resp, uuid);
+        cmd_resp::bind_term(&mut resp, self.term);
+        cb.call_box((resp,));
+
+        exec_result
+    }
+
+    fn clear_all_commands_as_stale(&mut self) {
+        while let Some(cmd) = self.pending_cmds.normals.pop_front() {
+            notify_stale_command(self.region_id, self.term, cmd);
+        }
+        if let Some(cmd) = self.pending_cmds.conf_change.take() {
+            notify_stale_command(self.region_id, self.term, cmd);
+        }
+    }
+
+    fn find_cb(&mut self, uuid: Uuid, term: u64, cmd: &RaftCmdRequest) -> Option<Callback> {
+        if get_change_peer_cmd(cmd).is_some() {
+            if let Some(mut cmd) = self.pending_cmds.take_conf_change() {
+                if cmd.uuid == uuid {
+                    return Some(cmd.cb.take().unwrap());
+                } else {
+                    notify_stale_command(self.region_id, self.term, cmd);
+                }
+            }
+            return None;
+        }
+        while let Some(mut head) = self.pending_cmds.pop_normal(term) {
+            if head.uuid == uuid {
+                return Some(head.cb.take().unwrap());
+            }
+            // Because of the lack of original RaftCmdRequest, we skip calling
+            // coprocessor here.
+            // TODO: call coprocessor with uuid instead.
+            notify_stale_command(self.region_id, self.term, head);
+        }
+        None
+    }
+
+    // apply operation can fail as following situation:
+    //   1. encouter an error that will occur on all store, it can continue
+    // applying next entry safely, like stale epoch for example;
+    //   2. encouter an error that may not occur on all store, in this case
+    // we should try to apply the entry again or panic. Considering that this
+    // usually due to disk operation fail, which is rare, so just panic is ok.
+    fn apply_raft_cmd(&mut self,
+                      db: Arc<DB>,
+                      index: u64,
+                      term: u64,
+                      req: &RaftCmdRequest)
+                      -> (RaftCmdResponse, Option<ExecResult>) {
+        // if pending remove, apply should be aborted already.
+        assert!(!self.pending_remove);
+
+        let mut ctx = ExecContext::new(self, db.clone(), index, term, req);
+        let (resp, exec_result) = self.exec_raft_cmd(&mut ctx).unwrap_or_else(|e| {
+            error!("[region {}] execute raft command err: {:?}",
+                   self.region_id,
+                   e);
+            (cmd_resp::new_error(e), None)
+        });
+
+        ctx.apply_state.set_applied_index(index);
+        if !self.pending_remove {
```

Comment:
Where is the destruction for this PeerMeta in the apply worker after a change conf happens to remove this peer?
It seems no destroy task is dispatched in store.